crates/router/src/core/payment_methods/cards.rs

```diff
@@ -1218,6 +1223,21 @@ pub async fn filter_payment_methods(
                         &payment_method_object.payment_method_type,
                     );
 
+                    let connector_variant = api_enums::Connector::from_str(connector.as_str())
+                        .into_report()
+                        .change_context(errors::ApiErrorResponse::InternalServerError)?;
```

Comment:
400 bad request invalid connector?

Reply:
right, shall i change_context to `ConnectorError::InvalidConnectorName` first and then to `ApiErrorResponse::InternalServerError` ?